cmd/gindoid/assetserver.go

```diff
@@ -0,0 +1,34 @@
+package main
+
+import (
+	"net/http"
+	"os"
+)
+
+type AssetFS struct {
+	fs http.FileSystem
+}
+
+func NewAssetFS(path string) AssetFS {
+	dir := http.Dir(path)
+	return AssetFS{dir}
+}
+
+// Open files but not directories. Disallows directory listing of asset directories.
+func (as AssetFS) Open(path string) (http.File, error) {
+	fp, err := as.fs.Open(path)
+	if err != nil {
+		return nil, err
+	}
+
+	if os.IsNotExist(err) {
```

Comment:
The `err` at this point will always be `nil` since we already return above if the `err` is not nil.